### PR TITLE
Adding visualizations to convex optimization problems, from the cvxpy library

### PR DESCRIPTION
**Pull request by BarGoldman**
During the development process, we have created three different functions that allow to visualize the Problem expressions in a more accessible and clear way for library users.

### CODE REVIEW

File: .idea/modules.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (SteveDiamond) [inline .idea/modules.xml]: Probably all these .xml files shouldn't be here?

File: Visual/test.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
+"""
```
Comment (SteveDiamond) [inline Visual/test.py]: This is an interesting way to test. Normally we use pytest.

File: Visual/visual.py [hunk 3]
```diff
@@ -0,0 +1,664 @@
...
+
+        # Replace the variable name in the expression with 'x'
+
+        # Replace '^' with '**'
+
+        # Construct the lambda function and return the input for plot_function
+        func = lambda x: eval(expr)
+        return (func, -10, 10)
+
+    def draw_graph(self):
```
Comment (SteveDiamond) [inline Visual/visual.py]: It should be possible here to change the x limits, so it's not always -10 to 10.

**Review by SteveDiamond — APPROVED** [review]
This all looks great! Thanks for working on this! It's definitely helpful to have more visualization tools.

File: Visual/visual.py (hunk 3 above)
Comment (nivmoti) [reply Visual/visual.py]: we change it, now you can change the x limits when -10 to 10 is the default

**Comment by einav242** [comment]
Hi,

As you can see, we have updated the tests in the testpy.py file to be of type pytest. In the draw_graph function, we made a modification so that the range can be selected, and if no range is chosen, the default range will be set to -10,10.

**Comment by SteveDiamond** [comment]
LGTM! Could you remove the test1 folder and rename ``Visual`` as ``visual``? After that happy to merge.

**Comment by einav242** [comment]
Hi,
We have deleted the "test1" folder and changed "Visual" to "visual" as requested.

**Comment by SteveDiamond** [comment]
Nice job!

**Pull request merged**